### PR TITLE
HW Func count elements 

### PR DESCRIPTION
**Pull request by liza459**

### CODE REVIEW

File: src/FuncHW/Program.cs [hunk 1]
```diff
@@ -27,14 +27,21 @@ static void Main()
 
             //Func*** funcPersonIsActive = ***;
             //Func*** funcPersonPhoneNumberStartsWith7 = ***;
-            //Func*** funcPersonIsChild = ***;
+
+            Func<Person, bool> funcPersonIsChild = (Person person) => DateTime.Now.Year - person.BirthDate.Year > 17;
```
Comment (TIove) [inline src/FuncHW/Program.cs]: If person is child, I think it should be lees than 17 ?

File: src/FuncHW/Program.cs [hunk 2]
```diff
@@ -27,14 +27,21 @@ static void Main()
...
             //Func*** funcPersonHasShortName = ***;
 
-            //Func*** funcCatIsDomestic = ***;
+            Func<Cat, bool> funcCatIsDomestic = (Cat cats) => cats.IsDomestic;
+
             //Func*** funcCatColorIsDark = ***;
             //Func*** funcCatNameConteinsU = ***;
             //Func*** funcCatIsDomesticAndWhite = ***;
 
+            int res1 = ListExtensions.CountElements(persons, funcPersonIsChild);
```
Comment (TIove) [inline src/FuncHW/Program.cs]: Name variables according to value it contains. e.x. `countPersonIsChild` or `countChildrenInPersons` etc.

File: src/FuncHW/Program.cs [hunk 3]
```diff
@@ -27,14 +27,21 @@ static void Main()
...
+            int res2 = ListExtensions.CountElements(cats, funcCatIsDomestic);
+            Console.WriteLine($"Количество персон старше 17 лет: {res1}\n" +
```
Comment (TIove) [inline src/FuncHW/Program.cs]: Blank line before `WriteLine`

File: src/FuncHW/ListExtensions.cs [hunk 4]
```diff
@@ -10,6 +10,20 @@ public static class ListExtensions
...
+            List<T> resultList = new();
+            foreach (T item in values)
+            {
+                if (func(item))
+                {
+                    resultList.Add(item);
+                }
+            }
+            int count = resultList.Count;
+            return count;
```
Comment (TIove) [inline src/FuncHW/ListExtensions.cs:23]: Blank line before `return`

File: src/FuncHW/ListExtensions.cs [hunk 5]
```diff
@@ -9,7 +9,19 @@ public static class ListExtensions
...
```
Comment (vorobevamv) [inline src/FuncHW/ListExtensions.cs:23]: you can write it (22+23) in one line
return resultList.Count;

File: src/FuncHW/Program.cs [hunk 6]
```diff
@@ -27,16 +27,21 @@ static void Main()
...
             //Func*** funcPersonIsActive = ***;
             //Func*** funcPersonPhoneNumberStartsWith7 = ***;
-            //Func*** funcPersonIsChild = ***;
+
+            Func<Person, bool> funcPersonIsChild = (Person person) => DateTime.Now.Year - person.BirthDate.Year > 17;
+
             //Func*** funcPersonHasShortName = ***;
 
-            //Func*** funcCatIsDomestic = ***;
+            Func<Cat, bool> funcCatIsDomestic = (Cat cats) => cats.IsDomestic;
```
Comment (vorobevamv) [inline src/FuncHW/Program.cs]: here is one cat, not list or other collection; so, 'cat', not 'cats'

**Review by MorozVl — APPROVED** [review]

File: src/FuncHW/Program.cs [hunk 7]
```diff
@@ -27,16 +27,22 @@ static void Main()
...
             //Func*** funcCatColorIsDark = ***;
             //Func*** funcCatNameConteinsU = ***;
             //Func*** funcCatIsDomesticAndWhite = ***;
 
-            //write result to variable
-            //check using debug and breakpoint
+            int countPersonIsChild = ListExtensions.CountElements(persons, funcPersonIsChild);
+            int countCatIsDomestic = ListExtensions.CountElements(cats, funcCatIsDomestic);
+
+            Console.WriteLine($"Количество персон старше 17 лет: {countPersonIsChild}\n" +
```
Comment (TIove) [inline src/FuncHW/Program.cs]: старше -> младше

**Review by TIove — APPROVED** [review]

**Review by MorozVl — APPROVED** [review]

**Review by vorobevamv — APPROVED** [review]

File: src/FuncHW/ListExtensions.cs [hunk 8]
```diff
@@ -9,7 +9,19 @@ public static class ListExtensions
...
+            foreach (T item in values)
+            {
+                if (func(item))
+                {
+                    resultList.Add(item);
+                }
+            }
+
+            return resultList.Count;
+        }
```
Comment (TheDioBrando) [inline src/FuncHW/ListExtensions.cs:24]: i think that creating new list is unnecessary. May be simple counter would be better

**Review by TheDioBrando — APPROVED** [review]

**Pull request merged**